### PR TITLE
test: Rewrite tests to use store_event instead of create_event

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

**Comment by dashed** [comment]
@lynnagara Just for my own understanding, what's the motivating reason to prefer `self.store_event` over `self.create_event`?

**Comment by lynnagara** [comment]
@dashed `create_event` is going to be removed soon, it creates an old style event that does not get inserted into Snuba. `store_event` is close to how a real event gets ingested.

File: tests/sentry/api/endpoints/test_group_user_reports.py [hunk 1]
```diff
@@ -36,7 +38,13 @@ def create_environment(self, project, name):
...
     def create_events_for_environment(self, group, environment, num_events):
         return [
-            self.create_event(group=group, tags={"environment": environment.name})
+            self.store_event(
+                data={
+                    "tags": {"environment": environment.name},
+                    "timestamp": iso_format(before_now(seconds=1)),
+                },
+                project_id=self.project.id,
+            )
```
Comment (fpacifici) [inline tests/sentry/api/endpoints/test_group_user_reports.py]: In the previous implementation we were creating a events for a specific gorup, now two groups would be created. Does the test still make sense ?

**Review by fpacifici — APPROVED** [review]

File: tests/sentry/api/endpoints/test_group_user_reports.py (hunk 1 above)
Comment (lynnagara) [reply tests/sentry/api/endpoints/test_group_user_reports.py]: Ah, good point. Put them all in the same group.

**Pull request merged**